### PR TITLE
[BUGFIX] Allow cached resources to have custom calls

### PR DESCRIPTION
**Pull request by e0ipso**
@amitaibu this is ready for review.

### CODE REVIEW

**Comment by amitaibu** [comment]
Can you explain a bit please :)

**Comment by e0ipso** [comment]
The cache decorator is proxying all the calls declared in the `ResourceInterface` to the decorated underlying resource. But it was not doing it for any of the methods of the parents of `ResourceInterface`.

With this code, any method that is not declared in the class will try to make that method call it in the decorated resource.

File: src/Plugin/resource/Decorators/ResourceDecoratorBase.php [hunk 1]
```diff
@@ -309,4 +309,26 @@ public function getUrl(array $options = array(), $keep_query = TRUE, RequestInte
...
   }
 
+  /**
+   * If any method not declared, then defer it to the decorated field.
+   *
+   * This decorator class is proxying all the calls declared in the
+   * ResourceInterface to the underlying decorated resource. But it is not
+   * doing it for any of the methods of the parents of ResourceInterface.
+   *
+   * With this code, any method that is not declared in the class will try to
```
Comment (amitaibu) [inline src/Plugin/resource/Decorators/ResourceDecoratorBase.php:319]: and if the method doesn't exist? Maybe we want to add a more descriptive  error message?

Comment (e0ipso) [inline src/Plugin/resource/Decorators/ResourceDecoratorBase.php:319]: If the method does not exist we are falling back to PHP's native behavior, which is a fatal error.

Something like

``` php
$decorator->validMethodInDecorator();
```

and

``` php
$decorator->validMethodInDecorated();
```

will yield expected results.

Something like:

``` php
$decorator->invalidMethod()
```

will also result in the expected behavior, which is a fatal error.

**Comment by amitaibu** [comment]
Had tight schedule with zendcon going on. Can you merge this, and I'll do a post merge review? Sorry :/

**Comment by e0ipso** [comment]
zendcon?? So jealous!!

**Pull request merged**